### DISCUSSION TITLE
Sessions

### DISCUSSION THREAD

**Discussion by mig824** [etcd-io/etcd#14195] [Q&A] [answered]
Can there be more than one Session (with the same lease) running at the same time?

**Comment by SimFG** [comment, marked as answer]
maybe you can try `concurrency.NewSession(cli, concurrency.WithLease(leaseId))`, like:
```
session, err := concurrency.NewSession(cli)
newSession, nerr := concurrency.NewSession(cli, concurrency.WithLease(session.Lease()))
```
see more detail:
https://github.com/etcd-io/etcd/blob/248f072370628fa2ad7c3d73e3935ba5636c0442/tests/integration/v3_election_test.go#L239-L282
